h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2269,34 +2269,34 @@ public void onCompletion(CountedCompleter caller) {
           _state.updateState(beta, gginfo);
         if (!_parms._lambda_search)
           updateProgress(true);
-        boolean converged = _state.converged();
+        boolean converged = _earlyStopEnabled ?false:_state.converged(); // GLM specific early stop.  Disable is early stop is enabled
         if (converged) Log.info(LogMsg(_state.convergenceMsg));
-        return !stop_requested() && !converged && _state._iter < _parms._max_iterations;
+        return !stop_requested() && !converged && _state._iter < _parms._max_iterations && !_earlyStop;
       }
     }
 
     public boolean progressHGLMGLMMME(double sumDiff2, double sumeta2, int iteration, boolean atGLMMME, GLMModel 
             fixedModel, GLMModel[] randModels, Frame glmmmeReturns, Frame hvDataOnly, double[] VC1, double[][] VC2,
                                       double[][] cholR, Frame augZ) {
-      boolean converged = (sumDiff2 < (_parms._objective_epsilon*sumeta2));
+      boolean converged = _earlyStopEnabled ?false:(sumDiff2 < (_parms._objective_epsilon*sumeta2));
       if (atGLMMME) {
         _state._iterHGLM_GLMMME++;
       } else {
           _state._iter++;
           updateProgress(fixedModel, randModels, glmmmeReturns, hvDataOnly, VC1, VC2, sumDiff2,
                   sumDiff2 / sumeta2, true, cholR, augZ);
       }
-      return !stop_requested() && !converged && (iteration < _parms._max_iterations);
+      return !stop_requested() && !converged && (iteration < _parms._max_iterations) && !_earlyStop;
     }
     
     public boolean progress(double [] beta, double likelihood) {
       _state._iter++;
       _state.updateState(beta,likelihood);
       if(!_parms._lambda_search)
         updateProgress(true);
-      boolean converged = _state.converged();
+      boolean converged = _earlyStopEnabled ? false:_state.converged();
```

Comment:
I see a pattern here, this code is repeated twice, should be put into a converged() method

Reply:
Fixed.